opensquirrel/passes/merger/general_merger.py

```diff
@@ -48,16 +50,57 @@ def compose_bloch_sphere_rotations(a: BlochSphereRotation, b: BlochSphereRotatio
 
     combined_phase = np.round(a.phase + b.phase, order_of_magnitude)
 
-    generator = b.generator if a.is_identity() else a.generator if b.is_identity() else None
-    arguments = b.arguments if a.is_identity() else a.arguments if b.is_identity() else None
+    if a.is_identity():
```

Comment:
I didn't change this code in this PR, did I?

What do you mean?
- To have another PR implementing correctly `is_identity`?
- To use `is_identity_up_to_a_global_pahse` in line 53 above?

Reply:
No you did not change this indeed. I just came across it and thought those conditions might be a bit too stringent.

I guess this method is used as a check, to see if a BSR is basically an identity, and therefore can be ignored. Since we, at the moment, do not take the global phase into account in our decompositions, I don't think that our check for whether something is an identity should than be exactly equal to identity (i.e. not differing by a global phase). 

So as part of this PR we could change it to:
```
 def is_identity(self) -> bool:
    # Checks if the Bloch sphere rotation is equal to identity (up to a global phase).
    return abs(self.angle) < ATOL   
```

But we could also do it in another PR (up to you).